rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -32,23 +37,43 @@ use crate::StorageError;
 pub struct AdmissionControlledS3Storage {
     storage: S3Storage,
     #[allow(clippy::type_complexity)]
-    outstanding_read_requests: Arc<
-        Mutex<
-            HashMap<
-                String,
-                Shared<
-                    Pin<
-                        Box<
-                            dyn Future<Output = Result<(Arc<Vec<u8>>, Option<ETag>), StorageError>>
-                                + Send
-                                + 'static,
-                        >,
-                    >,
-                >,
+    outstanding_read_requests: Arc<tokio::sync::Mutex<HashMap<String, InflightRequest>>>,
```

Comment:
changing to tokio mutex isn't a change that comes without consideration. given the importance of this component is there a way to restructure to avoid this?